content/docs/azimuth/l2/roller-tutorial.md

```diff
@@ -0,0 +1,177 @@
+This step must be performed whether you're using a fakezod or a live ship.
+
+First we need to start `%roller` and `%roller-rpc` (the agent used to send
+commands to `%roller` via HTTP). If you are not making use of a front-end, and
+instead will be accepting transactions to batch entirely from within Urbit,
+`%roller-rpc` is not necessary. We assume you will be using Bridge as a
+front-end.
+
+To start the agents, enter the following command in dojo: `|rein %base [&
+%roller] [& %roller-rpc]`. You should see the following response:
+
+```
+gall: installing %roller-rpc
+>   'init'
+gall: installing %roller
+>   %received-azimuth-state
+```
+
+Note that `%received-azimuth-state` only indicates that `%roller` successfully
+subscribed to `%azimuth`; it does not indicate that you successfully completed
+step 1.
+
+Next we need to configure `%roller`. `%roller` has several parameters that can
+be tweaked, such as how many transactions it will accept from a given ship in a
+given time period, how long the time period is, how much gas to use, exceptions
+to these rules for particular ships, etc. We will cover these in a moment, but
+first there are two mandatory settings that must be configured: the Ethereum
+node to submit batches to, and the private key for the wallet that will submit
+the transactions.
+
+First let's set the Ethereum node to which batches will be submitted, which we
+will presume is an Infura node. This does not need to be the same node given in
+step 1, but it is fine for them to be the same (assuming it is configured to
+accept transactions - some Infura nodes are set up only to make blockchain data
+available but do not participate in propagating transactions). Enter
+`:roller|endpoint 'https://MAINNET_INFURA_URL' %mainnet` into dojo.
+
+Next we set the private key for the Ethereum wallet from which batches built by
+`%roller` will be submitted. We *strongly emphasize* that this step has serious
+security implications. *`%roller` will be able to spend ETH in this wallet.*
+This is mandatory, of course, as the primary function of `%roller` is to submit
+batches of transactions to Ethereum, which requires ETH. If you downloaded
+`%roller` from someone other than Tlon and have not personally audited their
+code, you are putting the assets in this wallet at risk. Also keep in mind that
+while Urbit has undergone several security audits, some components such as the
+runtime have never been audited by a third party. We maintain a list of these
+audits [here](/audits), as well as an
+[FAQ](https://urbit.org/faq#how-secure-is-urbit) on the state of security in
+Urbit. Thus we recommend a fresh wallet generated specifically to be used by
+`%roller` and to only put the amount of ETH in it that you expect to need in the
+near future.
+
+With that being said, the command to set the private key for the wallet to be
+utilized by the roller is `:roller|setkey '0x1234567890abcdef'`, where `0x...`
+is the private key.
```

Comment:
We should add that it's highly recommended that the private key is only used for sending L2 batches. We have facilities for dealing with the addresses being used for something else, but this means that the roller will get out of sync with the expected L1 nonce, and this could cause delays in the sending batches

Reply:
@yosoyubik Sorry it has taken me so long to circle back to this.

I don't fully understand what you're saying.

> We have facilities for dealing with the address being used for something else,

By this I think you're saying that the roller can figure out the right nonce to use if the address is used for another purpose besides sending L2 transactions.

>  but this means that the roller will get out of sync with the expected L1 nonce, and this could cause delays in the sending batches

But that contradicts this. Are you saying that it can figure out the right nonce, but it will first try with the expected nonce and if that doesn't work, try the next nonce after some delay?

Reply:
Yeah, that second sentence is probably a bit of a worst-case-scenario so I don't think that using the L1 nonce for something else would cause delays—when the out-of-sync-nonce case is detected, after the nonce is updated, the batch is sent again right away—unless there's something else going like the private key not having enough eth... etc.